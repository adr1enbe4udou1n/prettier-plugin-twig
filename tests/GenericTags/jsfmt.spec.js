run_spec(__dirname, ["twig"], {
    twigMultiTags: [
        "nav,endnav",
        "switch,case,default,endswitch",
        "ifchildren,endifchildren",
        "cache,endcache"
    ]
});
