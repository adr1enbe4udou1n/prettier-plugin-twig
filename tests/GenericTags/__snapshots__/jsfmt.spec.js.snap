// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cache.twig - twig-verify > cache.twig 1`] = `
{% cache globally using key craft.some.rather.long.property.chain.request.path for 3 weeks %}
    {% for block in entry.myMatrixField %}
        <p>{{ block.text }}</p>
    {% endfor %}
{% endcache %}

{# prettier-ignore #}
{% cache globally using key craft.some.rather.long.property.chain.request.path for 3 weeks %}
    {% for block in entry.myMatrixField %}
        <p>{{ block.text }}</p>
    {% endfor %}
{% endcache %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% cache globally
    using
    key
    craft.some.rather.long.property.chain.request.path
    for
    3
    weeks
%}
    {% for block in entry.myMatrixField %}
        <p>
            {{ block.text }}
        </p>
    {% endfor %}
{% endcache %}

{# prettier-ignore #}
{% cache globally using key craft.some.rather.long.property.chain.request.path for 3 weeks %}
    {% for block in entry.myMatrixField %}
        <p>{{ block.text }}</p>
    {% endfor %}
{% endcache %}

`;

exports[`header.twig - twig-verify > header.twig 1`] = `
{%  header "Cache-Control: max-age=" ~ (expiry.timestamp - now.timestamp)  %}

{# prettier-ignore #}
{%  header "Cache-Control: max-age=" ~ (expiry.timestamp - now.timestamp)  %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% header 'Cache-Control: max-age=' ~ (expiry.timestamp - now.timestamp) %}

{# prettier-ignore #}
{%  header "Cache-Control: max-age=" ~ (expiry.timestamp - now.timestamp)  %}

`;

exports[`includeCssFile.twig - twig-verify > includeCssFile.twig 1`] = `
{% includeCssFile "/assets/css/layouts/" ~ entry.layout ~ ".css" %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% includeCssFile '/assets/css/layouts/' ~ entry.layout ~ '.css' %}

`;

exports[`nav.twig - twig-verify > nav.twig 1`] = `
{% nav entry in entries %}
    <li>
        <a href="{{ entry.url }}">{{ entry.title }}</a>
        {% ifchildren %}
            <ul>
                {% children %}
            </ul>
        {% endifchildren %}
    </li>
{% endnav %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% nav entry in entries %}
    <li>
        <a href="{{ entry.url }}">{{ entry.title }}</a>
        {% ifchildren %}
            <ul>
                {% children %}
            </ul>
        {% endifchildren %}
    </li>
{% endnav %}

`;

exports[`paginate.twig - twig-verify > paginate.twig 1`] = `
{% paginate craft.entries.section('blog').limit(10) as pageInfo, pageEntries %}

{% paginate craft.entries.section('blog').limit(10) as pageInfo, pageEntries, pageProperties %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% paginate craft.entries.section('blog').limit(10) as pageInfo, pageEntries %}

{% paginate craft.entries.section('blog').limit(10)
    as
    pageInfo,
    pageEntries,
    pageProperties
%}

`;

exports[`redirect.twig - twig-verify > redirect.twig 1`] = `
{% redirect "pricing" 301 %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% redirect 'pricing' 301 %}

`;

exports[`switch.twig - twig-verify > switch.twig 1`] = `
{% switch matrixBlock.type %}



    {% case "text" %}

        {{ matrixBlock.textField | markdown }}

    {% case "image" %}

        {{ matrixBlock.image[0].getImg() }}

    {% default %}

        <p>A font walks into a bar.</p>
        <p>The bartender says, “Hey, we don’t serve your type in here!”</p>

{% endswitch %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% switch matrixBlock.type %}
    {% case 'text' %}
        {{ matrixBlock.textField|markdown }}
    {% case 'image' %}
        {{ matrixBlock.image[0].getImg() }}
    {% default %}
        <p>
            A font walks into a bar.
        </p>
        <p>
            The bartender says, “Hey, we don’t serve your type in here!”
        </p>
{% endswitch %}

`;
